pkg/trace/sampler/remote_rates.go

```diff
@@ -40,40 +40,34 @@ type RemoteRates struct {
 	mu         sync.RWMutex // protects concurrent access to samplers and tpsTargets
 	tpsVersion uint64       // version of the loaded tpsTargets
 
-	stopSubscriber context.CancelFunc
-	exit           chan struct{}
-	stopped        chan struct{}
+	client  *remote.Client
+	exit    chan struct{}
+	stopped chan struct{}
 }
 
 func newRemoteRates(maxTPS float64) *RemoteRates {
 	if !features.Has("remote_rates") {
 		return nil
 	}
-	remoteRates := &RemoteRates{
+	client, err := remote.NewClient(remote.Facts{ID: "trace-agent", Name: "trace-agent", Version: version.AgentVersion}, []data.Product{data.ProductAPMSampling})
+	if err != nil {
+		log.Errorf("Error when subscribing to remote config management %v", err)
+		return nil
```

Comment:
returning `nil` on purpose I guess?

Reply:
Checked with APM and they are ok with this. Since the err can't be bubbled up easily it's either this or a fatal